File: frameworks/bananabread/src/project.ts
import { Content, Resource } from "../seagreen/src/modules/io/index.ts";
import { Layout, type LayoutBuilder } from "../seagreen/src/modules/layouting/index.ts";
import { Service } from "../seagreen/src/modules/webservices/index.ts";
import { AsyncDatabase, Baseline, Crud, JsonService, Upload } from "./services.ts";

export function createApp(): LayoutBuilder {
  return Layout.create()
    .add("pipeline", Content.from(Resource.fromString("ok")))
    .add("baseline11", Service.from(Baseline))
    .add("baseline2", Service.from(Baseline))
    .add("upload", Service.from(Upload))
    .add("json", Service.from(JsonService))
    .add("async-db", Service.from(AsyncDatabase))
    .add("crud", Layout.create().add("items", Service.from(Crud)));
}

File: frameworks/fishcake/src/main/kotlin/com/httparena/fishcake/Project.kt
package com.httparena.fishcake

import com.httparena.fishcake.services.AsyncDatabase
import com.httparena.fishcake.services.Baseline
import com.httparena.fishcake.services.Crud
import com.httparena.fishcake.services.JsonService
import com.httparena.fishcake.services.Upload
import org.codegreen.modules.io.Content
import org.codegreen.modules.io.Resource
import org.codegreen.modules.layouting.Layout
import org.codegreen.modules.layouting.provider.LayoutBuilder
import org.codegreen.modules.webservices.addService

/**
 * Builds the routing tree, mirroring the C# `genhttp-11` entry's `Project.Create()`.
 *
 * Compression, static files, websockets and TLS/HTTP-2 from the original entry rely on
 * modules the CodeGreen port does not provide yet and are omitted; the implemented profiles
 * are baseline, pipelined, json, upload, async-db and crud.
 */
object Project {

    fun create(): LayoutBuilder =
        Layout.create()
            .add("pipeline", Content.from(Resource.fromString("ok")))
            .addService<Baseline>("baseline11")
            .addService<Baseline>("baseline2")
            .addService<Upload>("upload")
            .addService<JsonService>("json")
            .addService<AsyncDatabase>("async-db")
            .add("crud", Layout.create().addService<Crud>("items"))
}
